### PR TITLE
OE4T update and refactor

### PR DESCRIPTION
**Pull request by TheYoctoJester**
This PR brings the current state of the layers `kirkstone` branch, which also includes the `kas` based configurations. This specific branch contains the refactoring of the `meta-mender-tegra` directory from a layer into a directory containing layers for Jetpack 4 and Jetpack 5 based platforms, respectively.

The included state of the Jetpack 5 incarnation is directly based on https://github.com/OE4T/meta-mender-community/pull/17 to serve as a base for the ongoing development.

### CODE REVIEW

**Comment by dwalkes** [comment]
@TheYoctoJester I thought I could update our kirkstone branch to match https://github.com/mendersoftware/meta-mender-community/commits/kirkstone but it looks like your branch isn't rebased on the upstream and has different commit history.  Do you want to rebase on upstream?

I was originally thinking of targeting kirkstone-r35.x for work related to Jetpack 5 but I think given we can support both Jetpack 4 and 5 with your changes we may as well just use kirkstone.

**Comment by TheYoctoJester** [comment]
@dwalkes the missing rebase was an oversight, sorry. Should be fixed now.

File: meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-bsp/u-boot/patches/0003-Integration-of-Mender-boot-code-into-U-Boot.patch
Comment (dwalkes) [inline meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-bsp/u-boot/patches/0003-Integration-of-Mender-boot-code-into-U-Boot.patch]: I think we can safely remove all uboot patches and recipes from Jetpack 5 as uboot is no longer a part of Jetpack 5 support

File: meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-core/images/tegra-minimal-initramfs.bbappend
Comment (dwalkes) [inline meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-core/images/tegra-minimal-initramfs.bbappend]: I assume this one should be safe to remove in Jetpack 5 as well.

File: meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-core/initrdscripts/tegra-minimal-init_%.bbappend
Comment (dwalkes) [inline meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-core/initrdscripts/tegra-minimal-init_%.bbappend]: I assume we can remove the initrdscripts/tegra-minimal-init as well as I think this assumes cboot/tegra-boot-tools which also won't be a part of Jetpack 5.

File: meta-mender-tegra/meta-mender-tegra-jetpack5/scripts/manifest-tegra.xml
Comment (dwalkes) [inline meta-mender-tegra/meta-mender-tegra-jetpack5/scripts/manifest-tegra.xml]: @TheYoctoJester is this still used with the new kas scheme?

File: meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-mender/tegra-state-scripts/tegra-state-scripts_1.0.bb
Comment (dwalkes) [inline meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-mender/tegra-state-scripts/tegra-state-scripts_1.0.bb:20]: I think this is essentially the only part of this recipe to keep on the Jetpack 5 directory, the rest can be removed.

File: meta-mender-tegra/meta-mender-tegra-jetpack5/scripts/test/README.md
Comment (dwalkes) [inline meta-mender-tegra/meta-mender-tegra-jetpack5/scripts/test/README.md]: Probably should just delete this directory for now, won't work with Jetpack 5 as-is.  If we do add Jetpack 5 support we should probably find some way to share between 4 and 5 directories.

File: meta-mender-tegra/meta-mender-tegra-jetpack5/templates/bblayers.conf.sample
Comment (dwalkes) [inline meta-mender-tegra/meta-mender-tegra-jetpack5/templates/bblayers.conf.sample]: @TheYoctoJester are the templates here still used with the kas setup?  If so don't we need to change the layer reference for meta-mender-tegra?

File: meta-mender-tegra/meta-mender-tegra-jetpack5/templates/local.conf.append
Comment (dwalkes) [inline meta-mender-tegra/meta-mender-tegra-jetpack5/templates/local.conf.append]: I suspect this file can just be deleted but TX2 is not supported on Jetpack 5.

**Review by dwalkes — COMMENTED** [review]
@TheYoctoJester I've added some comments here.  I'm wondering about whether we should avoid adding files we know won't be useful with Jetpack 5 before this merge or delete after the merge.  Will wait for @madisongh to weigh in before I approve.

**Comment by TheYoctoJester** [comment]
@dwalkes thanks a lot! Will go through those one by one once I'm back the office, presumably Monday or Tuesday. I personally would like to remove as much of the old stuff as possible and not drag it along.
In most cases I just can't judge as I'm not super familiar with the Tegra stuff, hence need input like yours. Thanks!

**Comment by TheYoctoJester** [comment]
@dwalkes addressed the comments. One thing I also noticed is the `c-boot` appends in `recipes-bsp`, which are actually masked out in `layer.conf`. Guess they should also be removed?

**Comment by dwalkes** [comment]
@TheYoctoJester good catch, yes everything cboot should be out of the Jetpack 5 branch.

File: meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-bsp/tegra-binaries/mender-custom-flash-layout/jetson-nano-2gb-devkit/flash_mender.xml
Comment (dwalkes) [inline meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-bsp/tegra-binaries/mender-custom-flash-layout/jetson-nano-2gb-devkit/flash_mender.xml]: This file can be removed from Jetpack 5 since nano is no longer supported

File: meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-bsp/tegra-binaries/mender-custom-flash-layout/jetson-nano-devkit-emmc/flash_mender.xml
Comment (dwalkes) [inline meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-bsp/tegra-binaries/mender-custom-flash-layout/jetson-nano-devkit-emmc/flash_mender.xml]: This file can be removed from Jetpack 5 since nano is no longer supported

File: meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-bsp/tegra-binaries/mender-custom-flash-layout/jetson-nano-devkit/flash_mender.xml
Comment (dwalkes) [inline meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-bsp/tegra-binaries/mender-custom-flash-layout/jetson-nano-devkit/flash_mender.xml]: This file can be removed from Jetpack 5 since nano is no longer supported

File: meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-bsp/tegra-binaries/mender-custom-flash-layout/jetson-nano-emmc/flash_mender.xml
Comment (dwalkes) [inline meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-bsp/tegra-binaries/mender-custom-flash-layout/jetson-nano-emmc/flash_mender.xml]: This file can be removed from Jetpack 5 since nano is no longer supported

File: meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-bsp/tegra-binaries/mender-custom-flash-layout/jetson-nano-qspi-sd/flash_mender.xml
Comment (dwalkes) [inline meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-bsp/tegra-binaries/mender-custom-flash-layout/jetson-nano-qspi-sd/flash_mender.xml]: This file can be removed from Jetpack 5 since nano is no longer supported

File: meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-bsp/tegra-binaries/mender-custom-flash-layout/tegra186/flash_mender.xml
Comment (dwalkes) [inline meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-bsp/tegra-binaries/mender-custom-flash-layout/tegra186/flash_mender.xml]: This file can be removed from Jetpack 5 since TX2 is no longer supported

**Review by dwalkes — COMMENTED** [review]
Noticed a few more items which can be removed from the Jetpack 5 layer as well

**Comment by TheYoctoJester** [comment]
@dwalkes cleaned up and rebased :-)

File: meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-bsp/uefi/edk2-firmware-tegra_%.bbappend
Comment (maciej-narojczyk) [inline meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-bsp/uefi/edk2-firmware-tegra_%.bbappend:5]: Instead of patch we should use overlays

Comment (TheYoctoJester) [reply meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-bsp/uefi/edk2-firmware-tegra_%.bbappend:5]: > Instead of patch we should use overlays

Can you provide the change? This is directly based off the `naroj/dev3` branch.

Comment (maciej-narojczyk) [reply meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-bsp/uefi/edk2-firmware-tegra_%.bbappend:5]: Yes, I know and I will.  I marked this line quickly to not forget to do this. Sorry for confusion

Comment (dwalkes) [reply meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-bsp/uefi/edk2-firmware-tegra_%.bbappend:5]: @maciej-narojczyk if this is working for you the same way it was on your previous branch let's merge and you can propose additional PRs for additional changes.

**Review by dwalkes — APPROVED** [review]
Looks good to me, and thanks @TheYoctoJester and @maciej-narojczyk for getting this started.

File: meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-bsp/uefi/edk2-firmware-tegra_%.bbappend
Comment (maciej-narojczyk) [reply meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-bsp/uefi/edk2-firmware-tegra_%.bbappend:5]: ok

**Comment by maciej-narojczyk** [comment]
I see that kas build kas/jetson-agx-orin-devkit.yml is building core-image-minimal should we build demo-image-base instead ?

**Comment by TheYoctoJester** [comment]
@maciej-narojczyk the default is `core-image-minimal` to reduce to build time for the automated testing. But I'm open to suggestions. If the general concensus is that `demo-image-base` is the image to promote for people taking a first look, then I'm happy to take that. Thoughts @dwalkes @madisongh?

**Comment by TheYoctoJester** [comment]
@dwalkes as you approved, do you want to take this into the OE4T fork first and then PR back to the Mender one, or should I put that one directly in?

File: kas/include/tegra-base.yml
Comment (maciej-narojczyk) [inline kas/include/tegra-base.yml:32]: we should add this line here 
EXTRA_IMAGE_FEATURES ?= "debug-tweaks"

without it i couldnt login at all :)
what is login and password then ?

**Comment by maciej-narojczyk** [comment]
I got a working mender update with this PR and my local tweaks so it works.

**Comment by dwalkes** [comment]
> @dwalkes as you approved, do you want to take this into the OE4T fork first and then PR back to the Mender one, or should I put that one directly in?

I was originally thinking we'd stage all the Jetpack 5 changes, including the refactor to move Jetpack 4 into a subdirectory, in this OE4T repository, and only merge upstream when we've got Jetpack 5 support complete.  I'm open to other suggestions though.  If we needed to make any changes to Jetpack 4 support this would be another reason to merge now.  Perhaps the new kas changes might be a place where we'd want to make changes for Jetpack 4 which also go upstream.

> the default is core-image-minimal to reduce to build time for the automated testing. But I'm open to suggestions. If the general concensus is that demo-image-base is the image to promote for people taking a first look, then I'm happy to take that. Thoughts @dwalkes @madisongh?

I think demo-image-base probably gets a bit more love and is more functional but there's no reason I know about which means core-image-minimal won't work.

**Comment by dwalkes** [comment]
Merging here for the time being

File: meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-bsp/uefi/files/0009-BootChainDxe-fix.patch
Comment (dwalkes) [inline meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-bsp/uefi/files/0009-BootChainDxe-fix.patch:1]: @maciej-narojczyk not sure if you are following the progress in https://github.com/OE4T/meta-mender-community/pull/19#discussion_r1676842714 but I'm curious if you could shed any light there on the necessity/purpose of this patch and `0007-runtime-access-for-KernelCommandLine-efivar.patch`

**Pull request merged**
